math/bigint/src/biguint_ser.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use num_bigint::BigUint;
+use serde::Serialize;
+
+pub fn serialize<S>(int: &BigUint, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    let mut bz = int.to_bytes_be();
+
+    // Insert positive sign byte at start of encoded bytes if non-zero
+    if bz == vec![0] {
```

Comment:
Wonder if its cheaper to check  if `bz` == 1 and `bz[0]` == 0 instead of allocating a new vec and comparing...

Reply:
oops, i meant check if the length of bz == 1

Reply:
can just compare the vec with an array at same effect, which is what I just pushed